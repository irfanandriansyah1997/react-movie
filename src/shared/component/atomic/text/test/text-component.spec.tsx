import React from 'react';
import { mount } from 'enzyme';

import TextComponent from '../text.component';

describe('Testing <TextComponent> in atomic component ', () => {
    it('Generate text snapshot 1', () => {
        const card = mount(
            <TextComponent
                className="testing-generate-class-name"
                styling="default"
                color="#000"
                fontWeight={700}
                align="center"
            >
                Text Default With Black Color
            </TextComponent>
        );

        expect(card.html()).toMatchSnapshot();
    });

    it('Generate text snapshot 2', () => {
        const card = mount(
            <TextComponent
                className="testing-generate-class-name"
                styling="caption"
                color="#69acec"
                fontWeight={500}
                align="right"
            >
                Text Caption With Blue Color
            </TextComponent>
        );

        expect(card.html()).toMatchSnapshot();
    });

    it('Generate text snapshot 2', () => {
        const card = mount(
            <TextComponent
                className="testing-generate-class-name"
                styling="heading"
                color="#f0f0f0"
                fontWeight={400}
                align="left"
            >
                Text Heading With Whitesmoke
            </TextComponent>
        );

        expect(card.html()).toMatchSnapshot();
    });
});
