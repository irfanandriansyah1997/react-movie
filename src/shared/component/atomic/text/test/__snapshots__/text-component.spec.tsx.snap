// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TextComponent> in atomic component  Generate text snapshot 1 1`] = `"<p class=\\"text default testing-generate-class-name\\" style=\\"color: rgb(0, 0, 0); font-weight: 700; text-align: center;\\">Text Default With Black Color</p>"`;

exports[`Testing <TextComponent> in atomic component  Generate text snapshot 2 1`] = `"<p class=\\"text caption testing-generate-class-name\\" style=\\"color: rgb(105, 172, 236); font-weight: 500; text-align: right;\\">Text Caption With Blue Color</p>"`;

exports[`Testing <TextComponent> in atomic component  Generate text snapshot 2 2`] = `"<p class=\\"text heading testing-generate-class-name\\" style=\\"color: rgb(240, 240, 240); font-weight: 400; text-align: left;\\">Text Heading With Whitesmoke</p>"`;
